BHoMUpgrader/Upgrader.cs

```diff
@@ -378,12 +378,15 @@ private BsonDocument UpgradeObjectExplicit(BsonDocument document, Converter conv
                 WriteToLog("object updated: " + b.GetType().FullName);
                 BsonDocument newDoc = new BsonDocument(b);
 
-                // Copy over BHoM properties
-                string[] properties = new string[] { "BHoM_Guid", "CustomData", "Name", "Tags", "Fragments" };
-                foreach (string p in properties)
+                // Copy over BHoM properties if the old object was a BHoMObject (assumption around BHoM_Guid made here)
+                if (document.Contains("BHoM_Guid"))
```

Comment:
In principle Happy with this change, if we have made sure it does not have any degrading effect on previously explicitly versioned objects.

Reply:
Thanks @IsakNaslundBh - as mentioned in [this comment](https://github.com/BHoM/Versioning_Toolkit/pull/282#issuecomment-2657189989), I did the investigation and have not found any issues 👍